### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of undefined (reading 'onDidPerformUserAction')"

### DISCUSSION THREAD

**Discussion by sowhat13** [community/community#69669] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Hello, I can't connect to github copilot servers from VSC. I did not change anything but this problem occurs suddenly. I tried sign out and login again, I tried to restart my pc, restart my IDE, nothing worked, here is full output log

```
[INFO] [copilotIgnore] [2023-10-10T03:55:49.647Z] inactive
[INFO] [codequote] [2023-10-10T03:55:49.648Z] CodeQuote enabled
[ERROR] [default] [2023-10-10T03:55:49.651Z] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of undefined (reading 'onDidPerformUserAction')"
```
I didn't see anybody got same error like "Cannot read properties of undefined (reading 'onDidPerformUserAction')". Anything from other connect error topics not worked for me.

**Comment by vachmara** [comment, marked as answer]
> ### Select Topic Area
> Bug
> 
> ### Body
> Hello, I can't connect to github copilot servers from VSC. I did not change anything but this problem occurs suddenly. I tried sign out and login again, I tried to restart my pc, restart my IDE, nothing worked, here is full output log
> 
> ```
> [INFO] [copilotIgnore] [2023-10-10T03:55:49.647Z] inactive
> [INFO] [codequote] [2023-10-10T03:55:49.648Z] CodeQuote enabled
> [ERROR] [default] [2023-10-10T03:55:49.651Z] GitHub Copilot could not connect to server. Extension activation failed: "Cannot read properties of undefined (reading 'onDidPerformUserAction')"
> ```
> 
> I didn't see anybody got same error like "Cannot read properties of undefined (reading 'onDidPerformUserAction')". Anything from other connect error topics not worked for me.

@sowhat13 you probably need to update VSC. When you are executing VSC as administrator, updates are disabled. Updating VSC work perfectly for me.

--- 
I think this is related to this merged [Add new API proposal for chat onDidPerformUserAction event #183140
](https://github.com/microsoft/vscode/pull/183140)

**Reply by sowhat13** [reply]
Thank you, I wasn't aware updates are disabled when running as administrator. I redownloaded VSC from its website and overwrite on existing one, it worked fine.

**Reply by TranManhCuongQN** [reply]
I encountered the above error, but after I updated to VSCode version 1.85.0 and GitHub Copilot version 1.143.0, it worked. Try updating your version

**Comment by ghost** [comment]
I was also getting this exact error message in VSCode's Github Copilot output. Updating VSCode did not work, but signing out of github in VSCode and signing back in did. 

![image](https://github.com/community/community/assets/2751277/0811ead7-3df7-4617-b7b5-840804fe80d8)


Once signed out, I closed and reopened VSCode to get the copilot login prompt back.

**Reply by connordeekelly** [reply]
Thank you!! This worked for me.

**Comment by zoestoll** [comment]
None of the above suggestions worked for me. What did work was reloading the project directory. Close window, then re-open the project you were just in.

**Comment by vforvilela** [comment]
The previous recommendations didn't do the trick for me. 
The issue got resolved when I made the switch to the pre-release version.

**Comment by daniel542813x** [comment]
in my case was fix when  updated  vscode and  sign out of github

**Comment by Aternix12** [comment]
My fix was downloading the System installer version of visual studio code, because i have admin priveledges on windows by default (i know your not meant to but it makes my life easier). https://code.visualstudio.com/download.

**Comment by UcheEdeoga** [comment]
The Issue was resolved when I downloaded the latest version of VSCode. I ran into some errors trying to update. So I uninstalled the old version and installed the recent version.

**Comment by cesardka** [comment]
I was having this issue while on a Mac, OSX Monterey 12.6. I had to move VS Code out of the Download folder (I just moved it to Documents), and then I was finally able to update the IDE to the latest version and install Copilot without facing any errors

**Comment by maxiuw** [comment]
My student license expired and after i reactivated it, I could connect to copilot. I had to reinstall the vscode. Also remember to refresh it after you log in to github :)

**Comment by cristian495** [comment]
In my case the fix was as simple as logging out of GitHub and then logging back in again.
![image](https://github.com/community/community/assets/16846162/1faae36b-c8cf-4290-a68b-ac220d3372ea)
